cw-orch-daemon/src/env.rs

```diff
@@ -0,0 +1,133 @@
+            Some(parse_with_log(str_value, MIN_GAS_ENV_NAME))
+        } else {
+            None
+        }
+    }
+
+    /// Optional - Integer
+    /// Defaults to [`DEFAULT_TX_QUERY_RETRIES`]
+    /// This changes the number of tx queries before it fails if it doesn't find any result
+    pub fn max_tx_query_retries() -> usize {
+        if let Ok(str_value) = env::var(MAX_TX_QUERIES_RETRY_ENV_NAME) {
+            parse_with_log(str_value, MAX_TX_QUERIES_RETRY_ENV_NAME)
+        } else {
+            DEFAULT_TX_QUERY_RETRIES
+        }
+    }
+
+    /// Optional - Integer
+    /// Defaults to 1
+    /// Minimum block speed in seconds. Useful when the block speeds are varying a lot
+    pub fn min_block_speed() -> u64 {
+        if let Ok(str_value) = env::var(MIN_BLOCK_SPEED_ENV_NAME) {
+            parse_with_log(str_value, MIN_BLOCK_SPEED_ENV_NAME)
+        } else {
+            1
+        }
+    }
+
+    /// Optional - boolean
+    /// Defaults to "false"
+    /// Disable wallet balance assertion.
+    /// When balance assertion is enabled, it asserts that the balance of the sender is sufficient before submitting any transactions (during the simulation step)
+    pub fn disable_wallet_balance_assertion() -> bool {
+        if let Ok(str_value) = env::var(DISABLE_WALLET_BALANCE_ASSERTION_ENV_NAME) {
+            parse_with_log(str_value, DISABLE_WALLET_BALANCE_ASSERTION_ENV_NAME)
+        } else {
+            false
+        }
+    }
+
+    /// Optional - boolean
+    /// Defaults to "false"
+    /// Disable the "Enable Logs" message
+    /// It allows forcing cw-orch to not output anything
+    pub fn disable_logs_message() -> bool {
+        if let Ok(str_value) = env::var(DISABLE_LOGS_ACTIVATION_MESSAGE_ENV_NAME) {
+            parse_with_log(str_value, DISABLE_LOGS_ACTIVATION_MESSAGE_ENV_NAME)
+        } else {
+            false
+        }
+    }
+}
+
+/// Fetches the default state folder.
+/// This function should only error if the home_dir is not set and the `dirs` library is unable to fetch it
+/// This happens only in rare cases
+pub fn default_state_folder() -> Result<PathBuf, StdError> {
+    dirs::home_dir().map(|home| home.join(".cw-orchestrator"))
+        .ok_or( StdError::generic_err(
+            format!(
+                "Your machine doesn't have a home folder. You can't use relative path for the state file such as 'state.json'. 
+                Please use an absolute path ('/home/root/state.json') or a dot-prefixed-relative path ('./state.json') in the {} env variable.",
+                STATE_FILE_ENV_NAME
+            )))
+}
+
+fn parse_with_log<F: FromStr<Err = E>, E: std::fmt::Display>(
+    value: String,
+    env_var_name: &str,
+) -> F {
+    value
+        .parse()
+        .map_err(|e| {
+            StdError::generic_err(format!(
+                "Couldn't parse content of env var {env_var_name}, error : {e}"
+            ))
+        })
+        .unwrap()
```

Comment:
It's not really `cosmwasm_std::StdError` and we panic on error anyway, let's use `.expect()` or just `panic!()` here instead?